### ISSUE TITLE
More consistent formatting

### ISSUE THREAD

**Issue by fanzhuyifan** [chaotic-aur/kde-git-WIP-ALPHA#5]
Currently the shell scripts have inconsistent formatting, sometimes mixing spaces and tabs. It would be better if there is a consistent formatting applied.

However, I am not sure what is the best way to proceed here, since I couldn't find any automatic formatters for nushell after a quick search...

I guess switching to a more "standard" and better supported scripting language, like python, is also a possibility..

**Comment by alexjp** [comment]
> However, I am not sure what is the best way to proceed here, since I couldn't find any automatic formatters for nushell after a quick search...

yeah, this is due to very very recent improvements on nushell: https://www.nushell.sh/blog/2023-11-14-nushell_0_87_0.html#lsp-is-coming-to-nushell-toc

I switched from neovim to vscode to have the lsp. (I know... easy cope out, could have it setup on neovim, but... will do later, priorities :))

I am planning on taking an overview of what works and what not, and redo the scripts properly.( As an example, I was thinking of instead of doing all folders inside pkgs_*, to have a yaml/toml file and follow that file instead. this was mainly for ordering or disabling or trigger rebuild/bump in dependent packages. for now maybe I should clean that and only do it when I have a specific plan)

**Comment by fanzhuyifan** [comment]
> switched from neovim to vscode to have the lsp. (I know... easy cope out, could have it setup on neovim, but... will do later, priorities :))

Humm I already have the vscode-nushell-lang extension, but it does not appear to have formatting capabilities. Is this something different?

**Comment by alexjp** [comment]
> > switched from neovim to vscode to have the lsp. (I know... easy cope out, could have it setup on neovim, but... will do later, priorities :))
> 
> Humm I already have the vscode-nushell-lang extension, but it does not appear to have formatting capabilities. Is this something different?

there is this one: https://github.com/nushell/nufmt , but doesnt seem to work very well (at least when I tested it).

but the lsp is a big big win, no more `help` or `f1` in the nushell command line :)
